include/deal.II/base/mutable_bind.h

```diff
@@ -0,0 +1,320 @@
+  {
+    /**
+     * A typedef to the stored std::tuple type.
+     */
+    using TupleType = std::tuple<typename std::remove_cv<
+      typename std::remove_reference<FunctionArgs>::type>::type...>;
+
+    /**
+     * An std::function that stores the the original function.
+     */
+    const std::function<ReturnType(FunctionArgs...)> function;
+
+    /**
+     * Currently stored arguments. These are forwarded to the function object
+     * above, when calling operator()().
+     */
+    TupleType arguments;
+
+    /**
+     * Call the original function, passing as arguments the elements of the
+     * stored TupleType.
+     */
+    ReturnType
+    operator()() const;
+
+    /**
+     * Set the arguments to use in @p function, for next time
+     * operator()() is called.
+     */
+    void
+    set_arguments(const TupleType &value);
+
+    /**
+     * Parse the arguments to use in @p function from a string, for next time
+     * operator()() is called.
+     */
+    void
+    parse_arguments(const std::string &                           value_string,
+                    const std::unique_ptr<Patterns::PatternBase> &pattern =
+                      Patterns::Tools::Convert<TupleType>::to_pattern());
+
+    /**
+     * Set the arguments to use in @p function, for next time
+     * operator()() is called.
+     */
+    void
+    set_arguments(FunctionArgs... args);
+  };
+
+
+
+  /**
+   * Create a MutableBind object from a function pointer and a list of
+   * arguments.
+   *
+   * An example usage is given by:
+   * @code
+   * void my_function(const int &a, const double &b);
+   *
+   * auto bound = mutable_bind(my_function, 1, 2.0);
+   *
+   * bound(); // will execute my_function(1, 2.0);
+   *
+   * bound.set_parameters(2, 3.0);
+   * bound(); // will execute my_function(2, 3.0);
+   *
+   * bound.parse_parameters("3: 4.0");
+   * bound(); // will execute my_function(3, 4.0);
+   * @endcode
+   *
+   * @authors Luca Heltai, Matthias Maier, 2019.
+   */
+  template <typename ReturnType, class... FunctionArgs>
+  MutableBind<ReturnType, FunctionArgs...>
+  mutable_bind(ReturnType (*function)(FunctionArgs...),
+               typename identity<FunctionArgs>::type... arguments);
+
+  /**
+   * Create a MutableBind object from a function pointer, with uninitialized
+   * arguments.
+   *
+   * An example usage of this function is given by:
+   * @code
+   * void my_function(const int &a, const double &b);
+   *
+   * auto bound = mutable_bind(my_function);
+   *
+   * bound.set_parameters(2, 3.0);
+   * bound(); // will execute my_function(2, 3.0);
+   *
+   * bound.parse_parameters("3: 4.0");
+   * bound(); // will execute my_function(3, 4.0);
+   * @endcode
+   *
+   * Notice that, if you do not call one of the MutableBind::set_parameters()
+   * methods, or the MutableBind::parse_parameters(), the arguments passed to
+   * the function object will be initialized with the values coming from the
+   * arguments defalt constructors.
```

Comment:
```suggestion
   * arguments' default constructors.
```

Reply:
Maybe also say that this can only work if each argument type has a default constructor.